### PR TITLE
Config toggle to disable blockwise fusion

### PR DESCRIPTION
**Pull request by crusaderky**
Useful to debug / work around dask/dask#9888

### CODE REVIEW

**Comment by phofl** [comment]
I am not a fan of exposing this publicly, the reuse keys stuff goes away for DataFrames with dask-expr and I don't want to respect this option there. Disabling blockwise fusion kills the scheduler as soon as you want to scale out

File: dask/dataframe/tests/test_optimize_dataframe.py [hunk 1]
```diff
@@ -44,3 +46,48 @@ def test_optimize_blockwise():
     graph = optimize_blockwise(ddf.dask)
 
     assert len(graph) <= 4
+
+
+def test_blockwise_optimize_toggle():
```
Comment (crusaderky) [inline dask/dataframe/tests/test_optimize_dataframe.py:51]: I'm a bit confused as to why this test doesn't fail with dask-expr enabled?

Comment (phofl) [reply dask/dataframe/tests/test_optimize_dataframe.py:51]: This doesn't go through the dask-expr optimised, you'll end up with an un-optimized expression that is materialised and then goes through the dask/dask blockwise optimise

You would have to call ddf.optimize() if you want to go through the dask-expr optimiser. We should probably fix this FWIW in here as well

**Comment by fjetter** [comment]
I don't expect anyone to actually use this configuration. I think we do not have any kind of documentation about blockwise fusion so this public config is confusing. Even worse, I can't come up with a situation where I would recommend this to anybody. Even if a user is affected by https://github.com/dask/dask/issues/9888 there are other mitigation strategies that are generally more user friendly than disabling fusion.

I'm also not convinced that debugging this is such a common practice that we need a proper toggle for it. For dask devs, it should be sufficient to comment this stuff out, it's two lines after all.

**Pull request closed without merging**